cloud-functions/cf-emailer/src/main/java/org/broadinstitute/ddp/cf/AttachmentLoadedFunction.java

```diff
@@ -0,0 +1,141 @@
+import com.sendgrid.*;
+import org.apache.http.impl.client.HttpClients;
+
+import java.io.IOException;
+import java.nio.channels.Channels;
+import java.util.Map;
+import java.util.logging.Logger;
+import java.util.regex.Pattern;
+
+@SuppressWarnings("unused")
+public class AttachmentLoadedFunction implements BackgroundFunction<AttachmentLoadedFunction.GcsEvent> {
+    private static final Logger logger = Logger.getLogger(AttachmentLoadedFunction.class.getName());
+    private static final String ATTACHMENT_DISPOSITION = "attachment";
+    private static final String ATTACHMENT_PDF_TYPE = "application/pdf";
+    public static final String PATH_MAIL_SEND = "mail/send";
+
+    private static final String API_KEY = "SG_KEY";
+    private static final String FROM_EMAIL = "FROM_EMAIL";
+    private static final String FROM_NAME = "DEV";
+    private static final String TO_EMAIL = "TO_EMAIL";
+    private static final String TEMPLATE_ID = "d-5ec2c969bbad4fffabb8168abafa470c";
+    private static final String GCP_PROJECT = "broad-ddp-dev";
+
+    private final SendGrid sendGrid;
+    private final Storage storage;
+
+    public AttachmentLoadedFunction() {
+        GoogleCredentials googleCredentials;
+        try {
+            googleCredentials = GoogleCredentials.getApplicationDefault();
+        } catch (IOException e) {
+            throw new IllegalArgumentException("Cannot obtain GoogleCredentials", e);
+        }
+
+        storage = StorageOptions.newBuilder()
+                .setCredentials(googleCredentials)
+                .setProjectId(GCP_PROJECT)
+                .build()
+                .getService();
+
+        var httpClientBuilder = HttpClients.custom();
+        var client = new Client(httpClientBuilder.build());
+        this.sendGrid = new SendGrid(API_KEY, client);
+    }
+
+    @Override
+    public void accept(GcsEvent event, Context context) {
+        String bucketName = event.getBucket();
+        String filename = event.getName();
+        logger.info(String.format("%s triggered with EventID: %s. File %s created in bucket %s",
+                context.eventType(), context.eventId(), filename, bucketName));
+
+        Pattern pattern = Pattern.compile("([^/]+/[A-Z0-9]{10}).pdf");
+        if (!pattern.matcher(filename).matches()) {
+            logger.info("Filename pattern does not match, skipping");
+            return;
+        }
+
+        BlobId contentBlobId = BlobId.of(bucketName, filename);
+        Blob contentBlob = storage.get(contentBlobId);
+        BlobId metadataBlobId = BlobId.of(bucketName, pattern.matcher(filename)
+                .replaceAll("$1.metadata"));
+        Blob metadataBlob = storage.get(metadataBlobId);
+        Map<String, String> metadata = metadataBlob.getMetadata();
+        String originalFilename = metadata.get("meta_filename");
+        if (!contentBlob.getSize().equals(Long.valueOf(metadata.get("meta_filesize")))) {
+            throw new RuntimeException("File size mismatch");
```

Comment:
When we get these deployed to `dev`, we should make sure that logging errors and throwing exceptions will be picked up in stackdriver and show up in our Error Reporting so a human is notified of these issues.

Reply:
I'll check and let you know